akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExampleOptionStateWithFactoryB.scala

```diff
@@ -0,0 +1,92 @@
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.persistence.typed.scaladsl.PersistentBehaviors6.HandlerFactory
+import akka.persistence.typed.scaladsl.{Effect, PersistentBehaviors6}
+import docs.akka.persistence.typed.AccountExampleOptionStateWithFactoryB.AccountEntity.{Account, AccountCommand, AccountEvent}
+
+object AccountExampleOptionStateWithFactoryB {
+
+  object AccountEntity extends HandlerFactory[AccountCommand,  AccountEvent, Option[Account]]{
```

Comment:
This is the least interesting one. 

The state is encoded as Option[Account] on the factory and that creates more boilerplate because all functions are required to return Option now. 

This is the same as defining `CommandHandler[AccountCommand, AccountEvent, Option[Account]]`.

The Option type leaks everywhere.